Beam/resources/assets/js/remplib.js

```diff
@@ -549,6 +575,64 @@ remplib = typeof(remplib) === 'undefined' ? {} : remplib;
             if (0 === this.totalTimeSpent % logInterval) {
                 this.trackTimespent();
             }
+        },
+
+        actualProgress: function() {
+            var h = document.documentElement,
+                b = document.body;
+
+            return (h.scrollTop||b.scrollTop) / ((h.scrollHeight||b.scrollHeight) - h.clientHeight);
+        },
+
+        scrollProgressEvent: function () {
+            var a = this.trackedArticle,
+                d = { document: this.actualProgress },
```

Comment:
If this isn't real DOM `document`, it probably shouldn't be called that way. It might cause false expectations when used later. How about `page`?